### PR TITLE
Fix to GitHub issues 574 and 580

### PR DESCRIPTION
**Pull request by yitam**
This change is [Reviewable](https://reviewable.io/reviews/microsoft/msphpsql/592)

### CODE REVIEW

File: source/shared/core_results.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ namespace {
 
 // *** internal constants ***
 
-const int INITIAL_FIELD_STRING_LEN = 256;    // base allocation size when retrieving a string field
+const int INITIAL_FIELD_STRING_LEN = 1024;      // base allocation size when retrieving a string field
```
Comment (v-kaywon) [inline source/shared/core_results.cpp]: can you only increase the initial buffer size when column encryption is enabled? Also, if column encryption is enabled, I would love to blast the initial buffer size to 8000 (max size for varchar(n))

File: test/functional/pdo_sqlsrv/pdo_574_next_rowset.phpt [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    unset($stmt);
+
+    // run queries in a batch
+    $stmt = $conn->prepare("SELECT * FROM [$tableName]; SELECT artist FROM [cd_info]; SELECT * FROM [$tableName1]");
+    $stmt->execute();
+    
+    // fetch from $tableName
+    $row = $stmt->fetch(PDO::FETCH_NUM);
+    if ($row) {
+        echo(substr($row[0], 0, 15)) . PHP_EOL;
```
Comment (v-kaywon) [inline test/functional/pdo_sqlsrv/pdo_574_next_rowset.phpt]: this only echoes the first 10 characters, what if something has gone wrong in between? Would be great of you can compare it with `$phrase` instead

File: test/functional/pdo_sqlsrv/pdo_574_next_rowset.phpt [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+    // create two tables with max fields
+    $columns = array(new ColumnMeta('varchar(max)', 'col1'));
+    createTable($conn, $tableName, $columns);
+
+    $columns = array(new ColumnMeta('varchar(max)', 'col1'));
+    createTable($conn, $tableName1, $columns);
+
+    // insert one row to each table
+    $phrase = str_repeat('This is a test ', 250);
```
Comment (v-kaywon) [inline test/functional/pdo_sqlsrv/pdo_574_next_rowset.phpt]: if you're going to change the initial buffer size, this number will have to change as well

File: test/functional/pdo_sqlsrv/pdo_574_next_rowset.phpt [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    $row = $stmt->fetch(PDO::FETCH_NUM);
+    if ($row) {
+        fatalError("This is unexpected!\n");
+    }
+
+    echo "4. next result? " . PHP_EOL;
+    try {
+        $next = $stmt->nextRowset();
+    } catch (PDOException $e) {
+        echo $e->getMessage() . PHP_EOL;
```
Comment (v-kaywon) [inline test/functional/pdo_sqlsrv/pdo_574_next_rowset.phpt]: This is a regression in 5.1.1-preview (see https://github.com/Microsoft/msphpsql/issues/581). newRowset() should not return any error. Please use 5.1.0-preview for testing and see what the behavior should be. Oh.. that means your test will fail against the dev branch for now... maybe talk to David to see when he'll revert the change and merge it with dev?

File: test/functional/sqlsrv/sqlsrv_574_next_result.phpt [hunk 5]
```diff
@@ -0,0 +1,186 @@
+--TEST--
+GitHub issue 574 - Fetch Next Result Test
```
Comment (v-kaywon) [inline test/functional/sqlsrv/sqlsrv_574_next_result.phpt]: please change this test as well according to the comments in the PDO_SQLSRV equivalent test

File: source/shared/core_results.cpp (hunk 1 above)
Comment (yitam) [reply source/shared/core_results.cpp]: I think it's reasonable to simply use 8000 regardless of AE setting. We use this buffer size for varchar / char parameters with unknown size.

**Comment by yitam** [comment]
#574 and #580

File: source/shared/core_results.cpp [hunk 6]
```diff
@@ -45,7 +45,7 @@ namespace {
 
 // *** internal constants ***
 
-const int INITIAL_FIELD_STRING_LEN = 1024;      // base allocation size when retrieving a string field
+const int INITIAL_FIELD_STRING_LEN = 8000;      // base allocation size when retrieving a string field
```
Comment (v-kaywon) [inline source/shared/core_results.cpp]: Now that I think about it, changing this value for non-AE may cause a breaking change. Say a user is trying to fetch a string that is 5000 bytes long. Before this change, the user needs to call sqlsrv_next_result + sqlsrv_get_field many times to retrieve the complete string. With this change however, the user only needs to call sqlsrv_get_field once the retrieve the entire string. This seems like a good thing. But if the user has multiple sqlsrv_next_result + sqlsrv_get_field in their code, error will be thrown telling the user there's no more result left. Therefore, I think it's best to only change the buffer size for if column encryption is enabled. It's impossible to change the constant here though, you may have to go through the code, check where `INITIAL_FIELD_STRING_LEN` is used and change it there.
